### PR TITLE
Opt in number fields into fallback synthetic source when doc values are disabled

### PR DESCRIPTION
**Pull request by lkts**
Contributes to #109546.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapperTests.java [hunk 1]
```diff
@@ -400,10 +408,27 @@ protected NumberSyntheticSourceSupport(Function<Number, Number> round, boolean i
...
+        }
+
         @Override
         public SyntheticSourceExample example(int maxVals) {
             if (randomBoolean()) {
                 Tuple<Object, Object> v = generateValue();
+                if (preservesExactSource()) {
+                    var rawInput = v.v1();
+
+                    // Confusing as it is this code actually runs
```
Comment (kkrik-es) [inline test/framework/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapperTests.java:425]: Nit: I'd skip the "Confusing as it is" part, it's all confusing.. Same below.

**Review by kkrik-es — APPROVED** [review]
Nice, this is refreshingly simple - bar the test updates..

**Review by martijnvg — APPROVED** [review]
LGTM

File: test/framework/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapperTests.java (hunk 1 above)
Comment (lkts) [reply test/framework/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapperTests.java:425]: Writing that made me feel better but i'll remove it.

**Pull request merged**